src/interfaces/InjectedSenderWallet.ts

```diff
@@ -101,16 +91,15 @@ export interface RequestSignTransactionsParams {
 }
 
 interface InjectedSenderWallet {
-  init: (params: InitParams) => Promise<InitResponse>;
+  isSender: string,
   getAccountId: () => string;
   getRpc: () => Promise<GetRpcResponse>;
   requestSignIn: (
```

Comment:
I'm rather concerned about the current state of this method since the update. I'm not sure if I'm doing something wrong but simply running the following always resolves:
```ts
near.requestSignIn({ contractId: "guest-book.testnet", methodNames: [] })
  .then((x) => console.log("then", x))
  .catch((err) => console.log("catch", err.message));
```
It logs the following to console: `"then Please connect first"`. There's three issues with this really:
- What exactly do I need to connect first? My wallet is unlocked etc.
- This is really an error so I should really see the following in console: `"catch Please connect first"`.
- The return type is either a `string` or an `object` now?

Reply:
It looks like we can also pass `amount` as a parameter to this method?

Reply:
`Please connect first`: This is an unexpected behavior from the wallet which should be fixed in next release (say next Monday). There's a `connect()` API added previously which needs to be called before `requestSignIn()`, but I have suggested to remove that.

Reply:
> `Please connect first`: This is an unexpected behavior from the wallet which should be fixed in next release (say next Monday or Tuesday). There's a `connect()` API added previously which needs to be called before `requestSignIn()`, but I have suggested to remove that.

Ah that makes sense. I've now noticed the `connect` method commented out in the example project and when checking directly in the browser. The main things for me are the last 2 points. I'm worried we'll either get runtime errors because the types are mismatched or the Promise is resolving when in fact there's an error, so the wallet selector won't detect something has gone wrong.

> The `amount` here means the `allowance` of the function call key, which is 0.25N by default.

Sorry, I was mostly referring to the types for this. We should add the `amount` property to `RequestSignInParams` for completeness. It's possible we might want to add support for this when we address the `methodNames` option too

Reply:
I have added the parameter type of `amount`. It's a string in yoctoⓃ